akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/SqlServerOffsetStoreDao.scala

```diff
@@ -97,16 +98,21 @@ private[projection] class SqlServerOffsetStoreDao(
    * @param notInLatestBySlice not used in postgres, but needed in sql
    * @return
    */
-  override protected def deleteOldTimestampOffsetSql(notInLatestBySlice: Seq[String]): String = {
+  override protected def deleteOldTimestampOffsetSql(notInLatestBySlice: Seq[LatestBySlice]): String = {
     val base =
-      s"DELETE FROM $timestampOffsetTable WHERE slice BETWEEN ? AND ? AND projection_name = ? AND timestamp_offset < ?"
+      s"DELETE FROM $timestampOffsetTable WHERE slice BETWEEN @from AND @to AND projection_name = @projectionName AND timestamp_offset < @timestampOffset"
     if (notInLatestBySlice.isEmpty) {
       sql"$base"
     } else {
-      val listParam = notInLatestBySlice.map(_ => "?").mkString(",")
+
+      val values = (timestampOffsetBySlicesSourceProvider.minSlice to timestampOffsetBySlicesSourceProvider.maxSlice)
+        .map { i =>
+          s"@slice_$i"
```

Comment:
maybe worth keeping all these parameter names as short as possible, if they are sent to the server?
```
s"@s_$i"
```